oli/src/commands/profile.rs

```diff
@@ -0,0 +1,154 @@
+
+use anyhow::anyhow;
+use anyhow::Result;
+use opendal::Accessor;
+use opendal::services::azblob;
+
+const OLI_PROFILE_PREFIX: &str = "OLI_PROFILE_";
+
+pub struct OliProfileGroup {
+    pub name: String,
+    pub typ: String,
+    pub values: HashMap<String, String>,
+}
+
+pub struct OliAccessor {
+    pub name: String,
+    pub accessor: Box<dyn Accessor>,
+}
+
+impl OliAccessor {
+    pub fn new(name: String, accessor: Box<dyn Accessor>) -> Self {
+        Self {
+            name,
+            accessor,
+        }
+    }
+}
+
+impl OliProfileGroup {
+    pub fn parse(name: &str, kvs: HashMap<String, String>) -> Result<Self> {
+        if let Some(typ) = kvs.get("type") {
+            Ok(OliProfileGroup {
+                name: name.to_string(),
+                typ: typ.to_string(),
+                values: kvs,
+            })
+        } else {
+            Err(anyhow!("oli profile must contains type"))
+        }
+    }
+}
+
+pub(crate) fn build_accessors() -> Result<Vec<OliAccessor>> {
+    let profiles = get_oli_profiles_from_env();
+    let groups = parse_oli_profile_groups(profiles)?;
+
+    let mut ret = Vec::new();
+    for group in groups {
+        let accessor = build_accessor(&group)?;
+        let oli_accessor = OliAccessor::new(group.name.clone(), accessor);
+        ret.push(oli_accessor);
+    }
+    Ok(ret)
+}
+
+fn get_oli_profiles_from_env() -> Vec<(String, String)> {
+    env::vars()
+        .filter(|(k, _)| k.starts_with(OLI_PROFILE_PREFIX))
+        .collect::<Vec<(String, String)>>()
+}
+
+fn parse_oli_profile_groups(profiles: Vec<(String, String)>) -> Result<Vec<OliProfileGroup>> {
+    let mut groups: HashMap<String, HashMap<String, String>> = HashMap::new();
+    for (key, value) in profiles {
+        let (group, trimmed_key) = parse_oli_profile(key)?;
+        if !groups.contains_key(&group) {
+            groups.insert(group.clone(), HashMap::new());
+        }
+        groups
+            .get_mut(&group)
+            .unwrap()
+            .insert(trimmed_key.to_lowercase(), value);
+    }
+
+    let mut rets = Vec::new();
+    for (name, kvs) in groups {
+        let group = OliProfileGroup::parse(&name, kvs)?;
+        rets.push(group);
+    }
+
+    Ok(rets)
+}
+
+fn parse_oli_profile(key: String) -> Result<(String, String)> {
+    let name_value = key
+        .strip_prefix(OLI_PROFILE_PREFIX)
+        .ok_or_else(|| anyhow!("invalid oli profile prefix"))?
+        .split_once('_');
+
+    match name_value {
+        Some((name, value)) => Ok((name.to_string(), value.to_string())),
+        None => Err(anyhow!("split name and value error")),
+    }
+}
+
+fn build_accessor(group: &OliProfileGroup) -> Result<Box<dyn Accessor>> {
+    let json_str = serde_json::to_string(&group.values)?;
+    let scheme = opendal::Scheme::from_str(&group.typ)?;
+    match scheme {
+        opendal::Scheme::Azblob => {
+            let builder = serde_json::from_str::<azblob::Builder>(&json_str)?.build()?;
```

Comment:
Please use https://opendal.databend.rs/opendal/struct.Operator.html#method.from_iter instead of convert by json.

Reply:
fixed~